frontend/src/libs/hooks/use-app-form/use-app-form.hook.ts

```diff
@@ -13,35 +15,45 @@ import { type ValidationSchema } from '~/libs/types/types.js';
 
 type Parameters<T extends FieldValues = FieldValues> = {
   defaultValues: DeepPartial<T>;
-  validationSchema: ValidationSchema;
+  validationSchema?: ValidationSchema;
+  mode: Mode;
 };
 
 type ReturnValue<T extends FieldValues = FieldValues> = {
   control: Control<T, null>;
   errors: FieldErrors<T>;
   handleSubmit: UseFormHandleSubmit<T>;
-  handleReset: UseFormReset<T>;
+  reset: UseFormReset<T>;
 };
 
 const useAppForm = <T extends FieldValues = FieldValues>({
   validationSchema,
   defaultValues,
+  mode,
 }: Parameters<T>): ReturnValue<T> => {
+  let parameters: UseFormProps<T> = {
+    mode,
+    defaultValues,
+  };
+
+  if (validationSchema) {
+    parameters = {
+      ...parameters,
+      resolver: joiResolver(validationSchema),
+    };
+  }
   const {
     control,
     handleSubmit,
     formState: { errors },
-    reset: handleReset,
-  } = useForm<T>({
-    defaultValues,
-    resolver: joiResolver(validationSchema),
-  });
+    reset,
+  } = useForm<T>(parameters);
 
   return {
     control,
     errors,
     handleSubmit,
-    handleReset,
+    reset,
```

Comment:
should be handleReset

Reply:
i received this error when using handleReset: `Property 'handleReset' does not exist on type 'UseFormReturn<T, any>'.` so I checked useForm documentation and found reset method

https://react-hook-form.com/ts/#UseFormReturn

Reply:
what should I do with reset?